rasa_nlu_examples/classifiers/sparse_sklearn_intent_classifier.py

```diff
@@ -0,0 +1,275 @@
+from abc import abstractmethod
+import logging
+import os
+import typing
+import warnings
+from typing import Any, Dict, List, Optional, Text, Tuple, Type
+
+import numpy as np
+import scipy.sparse
+
+import rasa.shared.utils.io
+import rasa.utils.io as io_utils
+from rasa.shared.constants import DOCS_URL_TRAINING_DATA_NLU
+from rasa.nlu.classifiers import LABEL_RANKING_LENGTH
+from rasa.nlu.featurizers.featurizer import SparseFeaturizer
+from rasa.nlu.components import Component
+from rasa.nlu.classifiers.classifier import IntentClassifier
+from rasa.nlu.config import RasaNLUModelConfig
+from rasa.shared.nlu.constants import TEXT
+from rasa.nlu.model import Metadata
+from rasa.shared.nlu.training_data.training_data import TrainingData
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.constants import (
+    INTENT,
+    INTENT_NAME_KEY,
+    PREDICTED_CONFIDENCE_KEY,
+)
+
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    import sklearn
+
+
+class SparseSklearnIntentClassifier(IntentClassifier):
+    r"""Base class for intent classifiers using the sklearn framework with sparse features.
+    Note that all sparse features will be used, i.e. there is no filtering for specific
+    featurizers yet."""
```

Comment:
```suggestion
    featurizers."""
```

Reply:
No need to suggest a change in the future. I'm certainly open to a change, but I wouldn't want to make any promises.